geopy/geocoders/mapbox.py

```diff
@@ -0,0 +1,209 @@
+            page (https://www.mapbox.com/account/access-tokens).
+
+        :param str domain: base api domain for mapbox
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+            :param ssl_context:
+                See :attr:`geopy.geocoders.options.default_ssl_context`.
+        """
+        super(MapBox, self).__init__(
+            format_string=format_string,
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+        )
+        if api_key == '':
+            raise ConfigurationError('Must provide an api_key')
+        self.api_key = api_key
+        self.domain = domain.strip('/')
+        self.api = '%s://%s/geocoding/v5/mapbox.places' % (self.scheme, self.domain)
+
+    @staticmethod
+    def _format_bbox_param(bbox):
+        """
+        Format the bbox to something Mapbox understands.
+        """
+        return '%f,%f,%f,%f' % (bbox[0], bbox[1], bbox[2], bbox[3])
+
+    @staticmethod
+    def _format_proximity_param(proximity):
+        """
+        Format the proximity to something Mapbox understands.
+        """
+        return '%f,%f' % (proximity[0], proximity[1])
+
+    def _parse_json(self, json, exactly_one=True):
+        '''Returns location, (latitude, longitude) from json feed.'''
+        features = json['features']
+        if features == []:
+            return None
+
+        def parse_feature(feature):
+            location = feature['place_name']
+            place = feature['text']
+            longitude = feature['geometry']['coordinates'][0]
+            latitude = feature['geometry']['coordinates'][1]
+            return Location(location, (latitude, longitude), place)
+        if exactly_one:
+            return parse_feature(features[0])
+        else:
+            return [parse_feature(feature) for feature in features]
+
+    @staticmethod
+    def _coerce_point_to_string(point):
+        """
+        Do the right thing on "point" input. For geocoders with reverse
+        methods.
+        """
+        if isinstance(point, Point):
+            return ",".join((str(point.longitude), str(point.latitude)))
+        elif isinstance(point, (list, tuple)):
+            return ",".join((str(point[1]), str(point[2])))  # -altitude
+        elif isinstance(point, string_compare):
+            return ",".join(point.split(",")[::-1])
+        else:
+            raise ValueError("Invalid point")
+
+    def geocode(
+            self,
+            query,
+            exactly_one=True,
+            proximity=None,
+            country=None,
+            bbox=None,
+            timeout=DEFAULT_SENTINEL
```

Comment:
`timeout` should be the third positional argument (i.e. right after `exactly_one`), to match the signature of the base method: 

https://github.com/geopy/geopy/blob/0a3a60bd6f9cb570e346ffef7953babf630ae8a3/geopy/geocoders/base.py#L368

Also please add trailing commas everywhere for cleaner diffs in future.